### PR TITLE
refactor: rewrite http_next ops to use op2 macro

### PR DESCRIPTION
**Pull request by bartlomieju**
Ref #19915

### CODE REVIEW

**Comment by bartlomieju** [comment]
Blocked by update to `deno_core` which will add support for "smi" return types.

File: ext/http/http_next.rs [hunk 1]
```diff
@@ -397,24 +400,22 @@ pub fn op_http_set_response_header(
   resp_headers.append(name, value);
 }
 
-#[op(v8)]
-fn op_http_set_response_headers(
+#[op2]
+pub fn op_http_set_response_headers(
   scope: &mut v8::HandleScope,
```
Comment (mmastrac) [inline ext/http/http_next.rs:405]: This `scope` isn't used, which should allow this method to become fast.

I'll file a bug -- we should be getting a warning that this parameter is unused. I suspect our `#[allow(unused)]` is a bit liberal.

File: ext/http/http_next.rs [hunk 2]
```diff
@@ -1088,11 +1100,15 @@ impl Resource for UpgradeStream {
...
-pub fn op_can_write_vectored(state: &mut OpState, rid: ResourceId) -> bool {
+#[op2(fast)]
+pub fn op_can_write_vectored(
+  state: &mut OpState,
+  #[smi] rid: ResourceId,
+) -> bool {
   state.resource_table.get::<UpgradeStream>(rid).is_ok()
 }
 
+// TODO(bartlomieju): op2 doesn't want to handle `usize` in the return type
```
Comment (mmastrac) [inline ext/http/http_next.rs:1111]: In this case it can be #[bigint] usize or #[bigint] u64, I think.

File: ext/http/http_next.rs [hunk 3]
```diff
@@ -382,11 +385,11 @@ pub fn op_http_read_request_body(
...
 }
 
-#[op(fast)]
+#[op2]
 pub fn op_http_set_response_header(
-  slab_id: SlabId,
-  name: ByteString,
-  value: ByteString,
+  #[smi] slab_id: SlabId,
+  #[serde] name: ByteString,
```
Comment (mmastrac) [inline ext/http/http_next.rs:391]: We need to add `fast` `ByteString` support for `#[string]` parameters.

**Review by mmastrac — APPROVED** [review]
LGTM with one small change

File: ext/http/http_next.rs (hunk 1 above)
Comment (bartlomieju) [reply ext/http/http_next.rs:405]: I' m not sure I understand - the scope parameter is used 5 times in a loop so I believe this is okay?

File: ext/http/http_next.rs (hunk 2 above)
Comment (bartlomieju) [reply ext/http/http_next.rs:1111]: I believe in this case we should cast it to `u32` before returning. The JS side expects a `number`, not `BigInt`

File: ext/http/http_next.rs (hunk 3 above)
Comment (bartlomieju) [reply ext/http/http_next.rs:391]: That would be lovely

File: ext/http/http_next.rs (hunk 2 above)
Comment (mmastrac) [reply ext/http/http_next.rs:1111]: Filed https://github.com/denoland/deno_core/issues/107 to make this `Number`-safe.

File: ext/http/http_next.rs (hunk 3 above)
Comment (mmastrac) [reply ext/http/http_next.rs:391]: https://github.com/denoland/deno_core/issues/108

File: ext/http/http_next.rs (hunk 1 above)
Comment (mmastrac) [reply ext/http/http_next.rs:405]: @bartlomieju I missed the collapsed section and thought this was a bug. Resolved, PR looks good to me.

**Pull request merged**